### ISSUE TITLE
Kernel is loading and does not run

### ISSUE THREAD

**Issue by gVallverdu** [nteract/archived-desktop-app#1297]
Hello
I have just install nteract from the dmg on mac os 10.12.2. The nteract version is `0.0.15`.

I had a R version working with Rstudio and python installed with anaconda. Jupyter notebook works well and I love it. That is what I would like to try nteract.

If I launch nteract, I got a notebook like page but at the bottom left there is `Loading ... | not connected`. If I try for example to run `2+3` in the cell, there is a star and nothing append. After this, at the bottom left there is `ir | idle`. This last message appears when I run ctrl+enter.

Moreover, I think nteract does not find python because in langage menu, there is only R. Again, in File -> New, there is only R.

I got this behavior whatever the way I launch nteract, from GUI or from terminal.

Regards

**Comment by rgbkrk** [comment]
Hi @gVallverdu! Thanks for trying out nteract.

> If I launch nteract, I got a notebook like page but at the bottom left there is Loading ... | not connected. If I try for example to run 2+3 in the cell, there is a star and nothing append. After this, at the bottom left there is ir | idle. This last message appears when I run ctrl+enter.

I've been running into this with the R kernel too. I _think_ this stems from the R kernel not booting as fast as we expect and so it misses our `kernel_info_request`. In general we need to retry our initial `kernel_info_request` so that we can serve kernels that are slower to start more effectively. I could be wrong on this one for R -- either way I'd love to see this experience get better.

> Moreover, I think nteract does not find python because in langage menu, there is only R. Again, in File -> New, there is only R.

In order to have Python kernels be detected, they need to be installed in a similar manner as the R kernel:

```
python -m pip install ipykernel
python -m ipykernel install --user
```

This installs a kernelspec for Python. You can also run `ipykernel` with options, including the name you want to make the kernel (in case you want individual ones for conda environments).

In the long run once https://github.com/nteract/nteract/pull/1264 is finished up, there are small PRs ahead to detect and use the default Python kernel as well as conda environments.

**Comment by gVallverdu** [comment]
Hi,
I do not understand why do I  need to install ipykernel once time more ? I have already a version installed :

```
--> pip freeze | grep ipykernel
ipykernel==4.5.0
```

Maybe I need to do only the second command ? In order to "activate" something in my user directory ?

Actually, I have just test this and it work with python now. What does actually do that ipykernel command ?

And what python virtualenv is used by interact by default ?

Thanks

**Comment by rgbkrk** [comment]
That ipykernel command places a `kernel.json` file in the jupyter kernels directory. In a coming release we'll do detection of a python installation that doesn't include the `kernel.json`.

Closing as duplicate of #1327 and #1384 for now. Thanks again for trying nteract out!
